aws/credentials.go

```diff
@@ -144,12 +144,15 @@ func IAMCreds() CredentialsProvider {
 // configuration file.
 func ProfileCreds(filename, profile string, expiry time.Duration) (CredentialsProvider, error) {
 	if filename == "" {
-		u, err := user.Current()
-		if err != nil {
-			return nil, err
+		homeDir := os.Getenv("HOME") // *nix
+		if homeDir == "" {           // Windows
+			homeDir = os.Getenv("USERPROFILE")
+		}
+		if homeDir == "" {
+			return nil, errors.New("User home directory not found.")
 		}
 
-		filename = path.Join(u.HomeDir, ".aws", "credentials")
+		filename = path.Join(homeDir, ".aws", "credentials")
```

Comment:
I think this should be filepath.Join (which joins with os.PathSeparator) vs path.Join (which joins with '/')